src/content/operations/nat-traversal/_index.en.md

```diff
@@ -0,0 +1,415 @@
+
+```bash
+kubectl label node $GWA --kubeconfig A gateway.submariner.io/natt-discovery-port=4490
+kubectl label node $GWB --kubeconfig B gateway.submariner.io/natt-discovery-port=4490
+kubectl label node $GWC --kubeconfig C gateway.submariner.io/natt-discovery-port=4491
+kubectl label node $GWC --kubeconfig C gateway.submariner.io/udp-port=4501
+kubectl label node $GWD --kubeconfig D gateway.submariner.io/natt-discovery-port=4492
+kubectl label node $GWD --kubeconfig D gateway.submariner.io/udp-port=4502
+
+# restart the gateways to pick up the new setting
+for cluster in A B C D;
+do
+  kubectl delete pod -n submariner-operator -l app=submariner-gateway --kubeconfig $cluster
+done
+```
+
+{{% notice warning %}}
+If **HA** is configured on the on-premises clusters, **each gateway** behind the 2.1.1.1 router
+**should have a dedicated UDP port**. For example if we had 2 clusters and 2 gateways on each cluster, 4
+ports would be necessary.
+{{% /notice %}}
+
+#### Router port mapping
+
+Under this configuration it's important to map the UDP ports on the 2.1.1.1 router to the private
+IPs of the gateways.
+
+| External IP | Port | Internal IP   | Port | Protocol |
+|:------------|:-----|:--------------|:-----|:---------|
+| 2.1.1.1     | 4501 | 192.168.0.4   | 4501 | UDP      |
+| 2.1.1.1     | 4491 | 192.168.0.4   | 4491 | UDP      |
+| 2.1.1.1     | 4502 | 192.168.128.4 | 4502 | UDP      |
+| 2.1.1.1     | 4492 | 192.168.128.4 | 4492 | UDP      |
+
+{{% notice note %}}
+Without port mapping it's entirely possible that the connectivity will be established without
+issues, this happens due to the fact that route's NAT will not generally modify the src port
+of the outgoing UDP packets, and future packets arriving this port will be redirected to the
+internal IP which initiated connectivity, but if the 2.1.1.1 router randomizes the source port on
+NAT, or if other applications on the internal network were already using the 4501-4502,4491-4492
+ports, the remote ends would not be able to contact gateway C or D over the expected ports.
+{{% /notice %}}
+
+#### Alternative to port mapping
+
+If port mapping is not possible, we can enable a server/client model for connections
+where we designate the clusters with a dedicated public IP, or the clusters with the
+ability to get mapped ports, as servers. In this way the other gateways on the Submariner
+network  will attempt (and not expect) connections to the remote endpoints, if such endpoints
+are designated as **preferred servers**.
+
+```bash
+subctl join --kubeconfig A --preferred-server .... broker_info.subm
+subctl join --kubeconfig B --preferred-server .... broker_info.subm
+```
+
+This will mean that cluster A & B gateways will connect to each other negotiating
+which one will be server based on the Endpoint names, cluster C will connect to cluster A & B
+as a client, cluster D will connect to cluster A & B as a client, cluster C & D will connect
+normally.
+
+### Multiple on-premises sites
+
```

Comment:
Similar comments for this section as above.

Reply:
Ack I will do it manually

Reply:
Done.